BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -323,9 +326,58 @@ public void UpdatePost(Post newPost, Post oldPost)
             SavePost(newPost);
         }
 
-        public bool CheckSlugExists(string slug)
+        public void SaveFiles(List<IFormFile> files)
+        {           
+            foreach(var file in files)
+            {
+                if(file.Length > 0)
+                {                    
+                    using (Stream uploadedFileStream = file.OpenReadStream())
+                    {
+                        byte[] buffer = new byte[uploadedFileStream.Length];
+                        uploadedFileStream.Read(buffer, 0, buffer.Length);
+                        string name = CreateFileName(file.FileName);
+                        _fileSystem.WriteFile($"{UploadsFolder}\\{name}", buffer);
+                    }
+                }
+            }
+        }
+
+        public IEnumerable<string> GetFileNames()
+        {
+            IEnumerable<string> fileNames = _fileSystem.EnumerateFiles(UploadsFolder);
+            return fileNames;
+        }
+
+        private bool CheckFileNameExists(string fileName)
         {
-            return _fileSystem.FileExists($"{PostsFolder}\\{slug}.xml");
+            return _fileSystem.FileExists($"{UploadsFolder}\\{fileName}");
         }
+
+        private string CreateFileName(string fileName)
+        {
+            string tempName = fileName;
+            string[] elements = fileName.Split(".");
+            int count = 0;
+            while (CheckFileNameExists(tempName))
+            {
+                count++;
+                if (elements.Length > 1)
+                {
+                    tempName = $"{elements[0]}-{count}.{elements[1]}";
+                }
+                else
+                {
+                    tempName = $"{fileName}-{count}";
+                }
+            }
+            return tempName;
+        }
+
+        public byte[] GetFileData(string fileName)
+        {
+            return _fileSystem.ReadAllBytes($"{UploadsFolder}\\{fileName}");

```

Comment:
This should check whether the file exists first and handle that gracefully (either by signaling that it doesn't, or by the caller to this method handling the error).  Either way, make sure we don't end up with an HTTP 500 when we should be giving a 404.